android/src/com/mapswithme/maps/MwmActivity.java

```diff
@@ -678,10 +704,60 @@ private boolean closePositionChooser()
       hidePositionChooser();
       return true;
     }
+    return false;
+  }
 
+  private boolean closeSearchToolbar(boolean clearText, boolean stopSearch)
```

Comment:
I don't see any case where stopSearch is false. Is this parameter really needed?

Reply:
In `refreshSearchToolbar()` line 546, I call `closeFloatingToolbars(false, false)` which in turn calls this function with stopSearch to false. `refreshSearchToolbar()` is called on activity resume lifecycle event, so this allows hiding the search toolbar without stopping the search in case the app is in navigation or planning state (to keep the search wheel working). In case the app is not in navigation or planning, then the search toolbar is shown again.

Reply:
Sorry, by some reason my browser did not show that diff.